pyout/tests/test_tabular.py

```diff
@@ -599,6 +600,85 @@ def test_tabular_write_intervals_bold():
     assert eq_repr(fd.getvalue(), expected)
 
 
+@patch("pyout.tabular.Terminal", TestTerminal)
+def test_tabular_write_transform():
+    fd = StringIO()
+    out = Tabular(style={"val": {"transform": lambda x: x[::-1]}},
+                  stream=fd)
+    out(OrderedDict([("name", "foo"),
+                     ("val", "330")]))
+    out(OrderedDict([("name", "bar"),
+                     ("val", "780")]))
+
+    expected = ("foo 033\n"
+                "bar 087\n")
+    assert eq_repr(fd.getvalue(), expected)
+
+
+@patch("pyout.tabular.Terminal", TestTerminal)
+def test_tabular_write_transform_with_header():
+    fd = StringIO()
```

Comment:
```
+@patch("pyout.tabular.Terminal", TestTerminal)
 +def test_...():
 +    fd = StringIO()
```
seems to be a common pattern.  Should there may be a fixture `test_terminal` which would actually provide the fd.

Evil me would even monkey patch that Terminal constructor to create a pass to it that StringIO so that test code had no need for any "test specific" inputs (such as `stream=fd`).  But that might not be worth it indeed ;)

Reply:
Yeah, here and in a few other places I think the tests could use
fixtures.  Haven't thought much about the approach for patching the
terminal.  Anyway, it's on my radar.